rustls/src/client/client_conn.rs

```diff
@@ -130,7 +129,7 @@ pub trait ResolvesClientCert: Send + Sync {
 /// * [`ClientConfig::session_storage`]: the default stores 256 sessions in memory.
 /// * [`ClientConfig::alpn_protocols`]: the default is empty -- no ALPN protocol is negotiated.
 /// * [`ClientConfig::key_log`]: key material is not logged.
-pub struct ClientConfig<C> {
```

Comment:
I had to tighten this bound from `C` to `C: CryptoProvider` in order to have the `kx_groups` field rely on the associated `KeyExchange` type (and in turn, its associated `SupportedGroup` type). Changing this bound cascaded to a few other places (e.g. the `clone` impl). The same flavour of change happened in a couple other places.

Reply:
Yup, that isn't really avoidable and should be mostly fine.